Indicators/ParabolicStopAndReverseExtended.cs

```diff
@@ -0,0 +1,304 @@
+            _afMaxLong = afMaxLong; 
+        }
+
+        /// <summary>
+        /// Create a new Parabolic SAR Extended
+        /// </summary>
+        /// <param name="sarStart">The starting value for the Parabolic Stop and Reverse indicator</param>
+        /// <param name="offsetOnReverse">The offset value to be applied on reverse </param>
+        /// <param name="afStartShort">The starting acceleration factor for short positions</param>
+        /// <param name="afIncrementShort">The increment value for the acceleration factor for short positions</param>
+        /// <param name="afMaxShort">The maximum value for the acceleration factor for short positions</param>
+        /// <param name="afStartLong">The starting acceleration factor for long positions</param>
+        /// <param name="afIncrementLong">The increment value for the acceleration factor for long positions</param>
+        /// <param name="afMaxLong">The maximum value for the acceleration factor for long positions</param>
+        public ParabolicStopAndReverseExtended(decimal sarStart = 0.0m, decimal offsetOnReverse = 0.0m, 
+            decimal afStartShort = 0.02m, decimal afIncrementShort = 0.02m, decimal afMaxShort = 0.2m, 
+            decimal afStartLong = 0.02m, decimal afIncrementLong = 0.02m, decimal afMaxLong = 0.2m) 
+            : this($"SAREXT({sarStart},{offsetOnReverse},{afStartShort},{afIncrementShort},{afMaxShort},{afStartLong},{afIncrementLong},{afMaxLong})", 
+            sarStart, offsetOnReverse, afStartShort, afIncrementShort, afMaxShort, afStartLong, afIncrementLong, afMaxLong)
+        {
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples >= 2;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod => 2;
+
+        /// <summary>
+        /// Resets this indicator to its initial state
+        /// </summary>
+        public override void Reset()
+        {
+            _afShort = _afInitShort;
+            _afLong = _afInitLong;
+            base.Reset();
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The trade bar input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            // On first iteration we can’t produce an SAR value so we save the current bar and return zero
+            if (Samples == 1){
+
+                 _previousBar = input;
+                // Makes sense to return _sarInit when its non-zero
+                if (_sarInit != 0) 
+                    return Math.Abs(_sarInit);
```

Comment:
- `){` following the project code style, opening brackets `{` are always in a new line & always present, even for 1 line `if` like this one. Please apply for all other other cases 👍 
- Why `Math.Abs()` in this case, given this extended version can return negative values? If this is still required though, suggest performing `Math.Abs()` once when this variable is set with a why comment explaining it 👍